ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/DifficultyCalculatorTests.java

```diff
@@ -0,0 +1,161 @@
+@RunWith(Parameterized.class)
+public class DifficultyCalculatorTests {
+
+  private final String testFile;
+  private final ProtocolSchedule<?> protocolSchedule;
+
+  public DifficultyCalculatorTests(
+      final String testFile, final ProtocolSchedule<?> protocolSchedule) {
+    this.testFile = testFile;
+    this.protocolSchedule = protocolSchedule;
+  }
+
+  @Parameters(name = "TestFile: {0}")
+  public static Collection<Object[]> getTestParametersForConfig() throws IOException {
+    return List.of(
+        new Object[] {
+          "/BasicTests/difficultyMainNetwork.json",
+          MainnetProtocolSchedule.fromConfig(GenesisConfigFile.mainnet().getConfigOptions())
+        },
+        new Object[] {
+          "/BasicTests/difficultyRopsten.json",
+          MainnetProtocolSchedule.fromConfig(
+              GenesisConfigFile.fromConfig(
+                      Resources.toString(
+                          GenesisConfigFile.class.getResource("/ropsten.json"),
+                          StandardCharsets.UTF_8))
+                  .getConfigOptions())
+        },
+        new Object[] {
+          "/BasicTests/difficultyFrontier.json",
+          MainnetProtocolSchedule.fromConfig(
+              GenesisConfigFile.fromConfig("{\"config\": {\"frontierBlock\":0}}")
+                  .getConfigOptions())
+        },
+        new Object[] {
+          "/BasicTests/difficultyHomestead.json",
+          MainnetProtocolSchedule.fromConfig(
+              GenesisConfigFile.fromConfig("{\"config\": {\"homesteadBlock\":0}}")
+                  .getConfigOptions())
+        },
+        new Object[] {
+          "/BasicTests/difficultyByzantium.json",
+          MainnetProtocolSchedule.fromConfig(
+              GenesisConfigFile.fromConfig("{\"config\": {\"byzantiumBlock\":0}}")
+                  .getConfigOptions())
+        },
+        new Object[] {
+          "/BasicTests/difficultyConstantinople.json",
+          MainnetProtocolSchedule.fromConfig(
+              GenesisConfigFile.fromConfig("{\"config\": {\"constantinopleBlock\":0}}")
+                  .getConfigOptions())
+        },
+        new Object[] {
+          "/BasicTests/difficultyEIP2384.json",
+          MainnetProtocolSchedule.fromConfig(
+              GenesisConfigFile.fromConfig("{\"config\":{\"muirGlacierBlock\":0}}")
+                  .getConfigOptions())
+        },
+        new Object[] {
+          "/BasicTests/difficultyEIP2384_random.json",
+          MainnetProtocolSchedule.fromConfig(
+              GenesisConfigFile.fromConfig("{\"config\":{\"muirGlacierBlock\":0}}")
+                  .getConfigOptions())
+        },
+        new Object[] {
+          "/BasicTests/difficultyEIP2384_random_to20M.json",
+          MainnetProtocolSchedule.fromConfig(
+              GenesisConfigFile.fromConfig("{\"config\":{\"muirGlacierBlock\":0}}")
+                  .getConfigOptions())
+        });
+  }
+
+  @Test
+  public void testDifficultyCalculation() throws IOException {
```

Comment:
Can this test not be run in isolation? Getting NullPointerExceptions when running this alone through intellij. Is there a gradle task that downloads resources?

Reply:
How many, if its 3 NPEs You need to update your submodules, if it's more you need to run it via gradle or have IntelliJ add the reference tests resoureces to the classpath when running.